### DISCUSSION TITLE
Sonnet.sln does not build due to missing dlls?

### DISCUSSION THREAD

**Discussion by zbatool** [coin-or/Sonnet#12] [Q&A] [answered]
I am having trouble successfully building SONNET in Visual Studio 2019. Following the instructions in Install.txt I cloned the SONNET source code from github, downloaded Cbc 2.10.3 sources from which I copied the folders BuildTools, Cbc, Cgl, Clp, CoinUtils and Osi to where my SONNET repository is. When I open the Sonnet.sln under SONNET\MSVisualStudio\v16 and build it in Visual Studio it cannot find any of the dlls such as Sonnet.dll or SonnetWrapper.dll as shown below
![image](https://user-images.githubusercontent.com/89423386/134599928-80c3f30c-eafe-404d-9127-3d3aacfe5a0c.png)

I have tried to build with configuration set to Release/Debug/ReleaseParallel and x64. Not sure if I'm missing a step or need to do something differently to build and run the Sonnet examples..

**Comment by jhmgoossens** [comment, marked as answer]
This is because (contrary to the INSTALL.txt guidance) the latest sources of Sonnet have to be built against the latest COIN-OR sources for Cbc etc--not the 2.10.3.
The latest build of sonnet against the latest sources of Cbc etc. works fine--but one of the tests fails but can probably be ignored.

Unfortunately, the Cbc folks do not publish the full 'tarball' of the latest sources of Cbc and dependencies any longer.
So either 1) get all the Cbc and dependency sources from github, 2) use coinbrew, or 3) get _all_ the sources from the latest appveyor build. 
I'll try to fix the appveyor build so you can use 3).

**Reply by jhmgoossens** [reply]
See [artifacts on AppVeyor](https://ci.appveyor.com/project/coin-or/sonnet/build/artifacts) that include all required sources. This is described as "Option 1" in INSTALL.txt.

**Reply by zbatool** [reply]
Thanks a lot! That solved the issue!
